src/FSharpLint.Core/Rules/Conventions/Naming/UnnestedFunctionNames.fs

```diff
@@ -0,0 +1,26 @@
+module FSharpLint.Rules.UnnestedFunctionNames
+
+open FSharp.Compiler.Syntax
+open FSharpLint.Framework.Ast
+open FSharpLint.Framework.AstInfo
+open FSharpLint.Framework.Rules
+open FSharpLint.Rules.Helper.Naming
+
+let private getIdentifiers (args: AstNodeRuleParams) =
+    match args.AstNode with
+    | AstNode.Binding (SynBinding (_, _, _, _, _attributes, _, _, pattern, _, _, _, _)) ->
+        if isNested args args.NodeIndex then
+            Array.empty
+        else
+            let allEncompassingAccessibility = Accessibility.Public
+            getPatternIdents allEncompassingAccessibility (fun _accessibility pat -> getFunctionIdents pat) true pattern
```

Comment:
@webwarrior-ws I don't like the variable name `pat` because it's ambiguous; can you please rename it to pattern? if you want to save up on characters to avoid horizontal scrolling, rather use well-known shorthands, for example a11y instead of "accessibility"

BTW, what does "allEncompassing" mean?

Reply:
We already have `pattern` binding in the scope, so I decided that shadowing might be confusing in this case.
"allEncompassing" means that it encompasses or includes other accessibility levels, such as Internal and Private.

Reply:
Let's call it `innerPattern` then, and rename the long accessibility vars to use a11y